src/Waives.Http/WaivesClient.cs

```diff
@@ -125,19 +126,41 @@ public WaivesClient Login(string clientId, string clientSecret)
         /// <seealso cref="Document.Delete"/>
         public async Task<Document> CreateDocument(Stream documentSource)
         {
+            if (documentSource == null)
+            {
+                throw new ArgumentNullException(nameof(documentSource));
+            }
+
+            if (!documentSource.CanSeek)
+            {
+                // This Stream implementation from ASP.NET Core buffers streams larger than the specified threshold to
+                // a temporary file, up to the buffer limit. Furthermore, it is a seekable Stream, so we can query the
+                // Stream's length
+                documentSource = new FileBufferingReadStream(
+                    documentSource,
+                    1024 * 1024, /* 1MiB */
+                    30 * 1000 * 1000, /* 30 MB */
+                    Path.GetTempPath);
+
+                await documentSource.ReadAsync(new byte[1], 0, 1).ConfigureAwait(false);
```

Comment:
```suggestion
            if (!documentSource.CanSeek)
            {
                // This Stream implementation from ASP.NET Core buffers streams larger than the specified threshold to
                // a temporary file, up to the buffer limit. Furthermore, it is a seekable Stream, so we can query the
                // Stream's length
                documentSource = new FileBufferingReadStream(
                    documentSource,
                    1024 * 1024, /* 1MiB */
                    30 * 1000 * 1000, /* 30 MB */
                    Path.GetTempPath);

                try {
                    await documentSource.ReadAsync(new byte[1], 0, 1).ConfigureAwait(false);
                    documentSource.Seek(0, SeekOrigin.Begin);
                } catch (EndOfStreamException) {
                    throw new ArgumentException("The provided stream has no content.", nameof(documentSource));
                }

            } else if (documentSource.Length < 1)
            {
                throw new ArgumentException("The provided stream has no content.", nameof(documentSource));
            }
```

Reply:
The docs on `ReadAsync()` and the implementation of `FileBufferingReadStream` do not make any mention of an `EndOfStreamException`, but I can see it's an exception exposed as part of the .NET Standard 🤔 What is the scenario you're guarding against here?